test_util/convertECL.cpp

```diff
@@ -149,31 +149,44 @@ int main(int argc, char **argv) {
         return 0;
     }
 
+    std::map<std::string, std::string> to_formatted = {{".EGRID", ".FEGRID"}, {".INIT", ".FINIT"}, {".SMSPEC", ".FSMSPEC"}, 
+        {".UNSMRY", ".FUNSMRY"}, {".UNRST", ".FUNRST"}, {".RFT", ".FRFT"}};
 
+    std::map<std::string, std::string> to_binary = {{".FEGRID", ".EGRID"}, {".FINIT", ".INIT"}, {".FSMSPEC", ".SMSPEC"}, 
+        {".FUNSMRY", ".UNSMRY"}, {".FUNRST", ".UNRST"}, {".FRFT", ".RFT"}};
+        
     if (formattedOutput) {
-        if (extension.substr(1,1)=="X"){
+        
+        auto search = to_formatted.find(extension);
+    
+        if (search != to_formatted.end()){
+            resFile = rootN + search->second;     
+        } else if (extension.substr(1,1)=="X"){
             resFile = rootN + ".F" + extension.substr(2);
         } else if (extension.substr(1,1)=="S"){
             resFile = rootN + ".A" + extension.substr(2);
         } else {
-            resFile = rootN + ".F" + extension.substr(1);
+            std::cout << "\n!ERROR, unknown file type for input file '" << rootN + extension << "'\n" << std::endl;
+            exit(1);
```

Comment:
It is in my opinion generally better to throw an exception than to use `exit( )`.